oeps/oep-0018-bp-python-dependencies.rst

```diff
@@ -400,6 +400,8 @@ packages from URLs whenever possible:
   partially worked around via a post-processing script for the generated
   requirements files; an example can be found in `edx-platform`_ at
   ``scripts/post-pip-compile.sh``.
+* When installing a package from PyPI, pip will not pull requirements
+  from URLs. It will only pull requirements from PyPI.
```

Comment:
Thanks, I'd missed that point.  The [full story](https://github.com/pypa/pip/issues/4187) behind this is pretty long and winding, but maybe this note should just mention that this is for security reasons (not allowing PyPI packages to fetch code from URLs with mutable content).

Reply:
Added the part about mutable contents.